src/main/java/com/bookstore/database/DBConnection.java

```diff
@@ -0,0 +1,34 @@
+package com.bookstore.database;
+
+import com.bookstore.Main;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+
+public class DBConnection {
+    private DBConnection() {
+    }
+
+    public static Connection getConnection() throws DBConnectionException {
+        Properties prop = new Properties();
+        try (InputStream input = Main.class.getClassLoader().getResourceAsStream("config.properties")) {
+            if (input == null) {
+                System.out.println("Unable to find config.properties");
+                throw new DBConnectionException("Config file not found");
+            }
+
+            prop.load(input);
+            String url = prop.getProperty("db.url");
```

Comment:
I like that you externalized this configuration, but the values are not safe if stored in the config.properties file. The best option would be to take values from environment variables